tests/integration/targets/clickhouse_role/tasks/initial.yml

```
@@ -0,0 +1,54 @@
+####################################################################
+# WARNING: These are designed specifically for Ansible tests       #
+# and should not be used as examples of how to write Ansible roles #
+####################################################################
+
+# Test
+- name: Create role in check mode
+  register: result
+  check_mode: true
+  community.clickhouse.clickhouse_role:
+    state: present
+    name: test_role
+
+- name: Check the role was created
+  register: result
+  community.clickhouse.clickhouse_client:
+    execute: "SELECT name FROM system.roles WHERE name = 'test_role'"
+
```

```suggestion
- name: Check the role is not present
  ansible.builtin.assert:
    that:
    - result.result[0] == []


```
Let's check here what our query above returned. May not work exactly how i put it but the idea is clear, just check the integration test output